test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -5230,15 +5276,63 @@ public MockSiteData register(SiteAndWebExchange sawe) {
     public final String strListID;
     public final String strItemID;
 
-    public URLSegmentsExchange(String strURL, boolean getURLSegmentsResult,
-        String strWebID, String strBucketID, String strListID,
-        String strItemID) {
-      this.strURL = strURL;
-      this.getURLSegmentsResult = getURLSegmentsResult;
-      this.strWebID = strWebID;
-      this.strBucketID = strBucketID;
-      this.strListID = strListID;
-      this.strItemID = strItemID;
+    public URLSegmentsExchange(Builder builder) {
+      this.strURL = builder.strURL;
+      this.getURLSegmentsResult = builder.getURLSegmentsResult;
+      this.strWebID = builder.strWebID;
+      this.strBucketID = builder.strBucketID;
+      this.strListID = builder.strListID;
+      this.strItemID = builder.strItemID;
+    }
+
+    private static class Builder {
+      private String strURL;
```

Comment:
Most of these names are style violations. Let's not fix the URLSegmentsExchange field (or class) names in this CL, but new names in the Builder should be fixed, e.g., s/strURL/strUrl/g. Also, what does "str" refer to here? Is it some domain thing, or just Hungarian notation copied from Microsoft doc? We don't use Hungarian.

Reply:
Done. str means String. I tried to retain current names. Thanks for pointing out. Updated.